test/test_genome.py

```diff
@@ -476,76 +476,48 @@ def select_gene_indices_non_silent(mutation_rate, rng):
     assert not only_silent_mutations
 
 
-def test_permissible_values_hidden(rng_seed):
-    genome_params = {
-        "n_inputs": 2,
-        "n_outputs": 1,
-        "n_columns": 3,
-        "n_rows": 3,
-        "levels_back": 2,
-        "primitives": (cgp.Add, cgp.Sub, cgp.ConstantFloat),
-    }
-
-    genome = cgp.Genome(**genome_params)
-
-    # test function gene
-    gene_idx = 6  # function gene of first hidden region
-    allele = 0  #
-    region_idx = None  # can be any number, since not touched for function gene
-    permissible_func_gene_values = genome._determine_alternative_permissible_values_hidden_gene(
-        gene_idx, allele, region_idx
-    )
-    assert permissible_func_gene_values == [
-        1,
-        2,
-    ]  # function idx 1,2 (cgp.Sub, cgp.ConstantFloat)
-
-    # test input gene
-    gene_idx = 16  # first input gene in second column of hidden region
-    allele = 0
-    region_idx = gene_idx // (
-        genome.primitives.max_arity + 1
-    )  # function gene + input gene addresses
-    permissible_input_gene_values = genome._determine_alternative_permissible_values_hidden_gene(
-        gene_idx, allele, region_idx
-    )
-
-    assert permissible_input_gene_values == [
-        1,
-        2,
-        3,
-        4,
-    ]  # gene indices of input, 1st hidden layer
-
-
-def test_permissible_values_output(rng_seed):
-    genome_params = {
-        "n_inputs": 2,
-        "n_outputs": 1,
-        "n_columns": 3,
-        "n_rows": 3,
-        "levels_back": 2,
-        "primitives": (cgp.Add, cgp.Sub, cgp.ConstantFloat),
-    }
+def test_permissible_values(genome_params):
     genome = cgp.Genome(**genome_params)
+    permissible_values_expected = [
+        np.array(ID_INPUT_NODE),
+        np.array(ID_NON_CODING_GENE),
+        np.array(ID_NON_CODING_GENE),
+        np.array(ID_INPUT_NODE),
+        np.array(ID_NON_CODING_GENE),
+        np.array(ID_NON_CODING_GENE),
+        np.arange(len(genome._primitives._primitives)),
+        np.array([0, 1]),
+        np.array([0, 1]),
```

Comment:
Can you define some variable that represents `np.array([0,1])` etc. here so that you don't need to repeat this expression so many times? Ideally with a meaningful name that explains why the permissible values are 0,1 here, (and 0,1,2,3,4 elsewhere etc.).

Reply:
I tried to add meaningful names in the latest commit, please tell me if they are ok. Also i moved the genome params inside the function, since the permissible_values_expected depend strongly on the specific values.